### PR TITLE
Add pic variant to NCEPLIBS libraries

### LINKED ISSUE #260: Add PIC option to NCEPLIBS packages
Soon the saber repo will depend on https://github.com/GEOS-ESM/gsibclim, which will in turn depend on https://github.com/NOAA-EMC/NCEPLIBS-sp. So this NCEPLIBS-sp package should be added to the JEDI stack. We are currently testing this repo on Discover so it would be useful to have available there before too long.

### PR DESCRIPTION
**Pull request by kgerheiser**
Fix https://github.com/NOAA-EMC/spack-stack/issues/260

### CODE REVIEW

File: var/spack/repos/builtin/packages/g2c/package.py [hunk 1]
```diff
@@ -16,23 +16,22 @@ class G2c(CMakePackage):
 
     maintainers = ['kgerheiser', 'Hang-Lei-NOAA', 'edwardhartnett']
 
+    version('1.6.4', sha256='5129a772572a358296b05fbe846bd390c6a501254588c6a223623649aefacb9d')
+    version('1.6.2', sha256='b5384b48e108293d7f764cdad458ac8ce436f26be330b02c69c2a75bb7eb9a2c')
+
```
Comment (kgerheiser) [inline var/spack/repos/builtin/packages/g2c/package.py:21]: Just fixing the ordering here with versions on top

Comment (climbfuji) [reply var/spack/repos/builtin/packages/g2c/package.py:21]: Thanks!

**Review by climbfuji — APPROVED** [review]
Thanks for these changes. Do we need to do the same for the less commonly used NCEPLIBS?

**Comment by climbfuji** [comment]
Linux Style tests are failing ...

**Comment by kgerheiser** [comment]
I see, but it doesn't seem to be related to the style itself. Trying to figure out what's going wrong.

```
==> Running style checks on spack
[8](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:9)
  selected: isort, mypy, flake8
[9](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:10)
==> Modified files
[10](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:11)
  var/spack/repos/builtin/packages/bacio/package.py
[11](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:12)
  var/spack/repos/builtin/packages/g2/package.py
[12](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:13)
  var/spack/repos/builtin/packages/g2c/package.py
[13](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:14)
  var/spack/repos/builtin/packages/sp/package.py
[14](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:15)
  var/spack/repos/builtin/packages/w3emc/package.py
[15](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:16)
==> Error: cannot bootstrap any of the isort executables from spec "py-isort@4.3.5: %gcc target=x86_64" due to the following failures:
[16](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:17)
==> Bootstrapping py-isort from pre-built binaries
[17](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:18)
==> Bootstrapping py-isort@4.3.5: %gcc target=x86_64 from sources
[18](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:19)
==> Bootstrapping clingo from pre-built binaries
[19](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:20)
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-2l45t4kw3cqqwj6vbxhfwhzlo6b3q2p4.spec.json
[20](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:21)
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64/gcc-10.2.1/clingo-bootstrap-spack/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-2l45t4kw3cqqwj6vbxhfwhzlo6b3q2p4.spack
[21](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:22)
==> Installing "clingo-bootstrap@spack%gcc@10.2.1~docs~ipo+python+static_libstdcpp build_type=Release arch=linux-centos7-x86_64" from a buildcache
[22](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:23)
github-actions-v0.2 raised FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/spack/spack/share/spack/bootstrap/github-actions-v0.2/py-isort.json'
[23](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:24)
github-actions-v0.1 raised ValueError: source is not trusted
[24](https://github.com/NOAA-EMC/spack/runs/7431994233?check_suite_focus=true#step:6:25)
spack-install raised UnknownPackageError: Package 'pkgconfig' not found.
```

**Comment by kgerheiser** [comment]
I had a mis-indented line in one of the packages and somehow that breaks that bootstrap process for py-isort in the style check. Who knew?

**Comment by kgerheiser** [comment]
@climbfuji ready for review again

**Review by climbfuji — APPROVED** [review]

**Pull request merged**